app/controllers/network_routes_controller.go

```diff
@@ -1276,7 +1278,11 @@ func (nrc *NetworkRoutingController) OnNodeUpdate(nodeUpdate *watchers.NodeUpdat
 	if nrc.bgpEnableInternal {
 		nrc.syncInternalPeers()
 	}
-	nrc.disableSourceDestinationCheck()
+
+	// skip if first round of disableSourceDestinationCheck() is not done yet
+	if nrc.disableSrcDstCheckDone {
```

Comment:
Is this supposed to be `if !nrc.disableSrcDstCheckDone`?

Reply:
logic is correct, but naming is very poor. I will correct it.

Basically when kube-rotuer pod starts, master nodes try to disable src-dst check for all the nodes just one time.

Until its done, dont try to do it again for node updates as well.

i guess whats happening was, you launch cluster with 30-40 nodes for e.g, then for each node update, same task was being done concurrently so ec2 api serves was throttling the requests.

this check prevents unnecessary calls while in the initial (cluster booting) phase.